syntax = "proto3";

package resembla.server;

message ResemblaRequest {
  string query = 1;
}

message ResemblaOnDemandRequest {
  string query = 1;
  repeated string candidates = 2;
}

message ResemblaResponse {
  message Result {
    int32 id = 1;
    string text = 2;
    float score = 3;
  }
  repeated Result results = 1;
}

service ResemblaService {
  rpc find (ResemblaRequest) returns (ResemblaResponse) {}
  rpc eval (ResemblaOnDemandRequest) returns (ResemblaResponse) {}
}
